### PR TITLE
[RHDX-67] Add Drupal caching to Product downloads

### PR DESCRIPTION
**Pull request by jordanpagewhite**
This adds a Drupal cache bin and drupal_static() caching to our Download
Manager interactions. This should significantly improve the performance
of, most notably, the product download pages, but more generally pages
that reference Product Download Hero assembl(ies) and/or Product
Download List assembl(ies).

### JIRA Issue Link
* Related to: https://projects.engineering.redhat.com/browse/RHDX-67

#### 'download_manager_api:product:' . $id cache bin

* This implements a cache bin for Download Manager responses by Product ID.

TODO: We will need to implement a configurable TTL in the future. It is currently set, in code, to an hour

#### drupal_static() caching

* This will provide static caching of the Download Manager response, getProductById() method. This should considerably improve the performance of Product download pages, almost all of which make multiple getProductById() calls i.e. Product Download Hero + Product Download List

##### See

https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/drupal_static/8.2.x
https://www.oreilly.com/library/view/high-performance-drupal/9781449358013/ch04.html (static caching section)

#### Product fileSize validation

* The Product Download List twig file should now validate that the fileSize exists before trying to render the value. If not value exists, it will just render 'Download'

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php [hunk 1]
```diff
@@ -33,18 +36,55 @@ class DownloadManagerApi {
...
+
   /**
    * Constructor.
    *
    * @param GuzzleHttp\ClientInterface $client
    *   The Guzzle client.
    * @param Drupal\Core\Entity\EntityTypeManager $entity_type_manager
-   *   Drupal's entity type manager.
+   *   entity type manager.
+   * @var \Drupal\Core\Cache\CacheBackendInterface $cache_bin
```
Comment (staceymosier) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php]: Considering phpcs integration:

There are PHPCS warnings when this is @var, as well as the next 2.  The warnings are cleared when changing this from @var to @param. 

Line 117 exceeds the limit (80 characters) by 1.

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

**Pull request merged**